### PR TITLE
Mock version and source map in plugin snapshot tests

### PR DESCRIPTION
**Pull request by tomekzaw**
## Summary

I noticed that `version` and `sourceMap` properties are not included in plugin snapshots. This PR:

* renames environmental variables used for testing purposes
* unifies `compact` setting across app and tests and updates snapshots accordingly
* restores `version` property in snapshots but uses mock version "x.y.z" instead of real version
* moves `version` property to `__initData` to prevent allocating multiple strings (see #3930 for explanation)
* restores `sourceMap` property in `__initData` but uses mock source map for all snapshot tests
* adds one unit test that checks real `sourceMap` property
* adds double negation to fix return type of `isRelease` (previously it returned `boolean | string | ""`)

### CODE REVIEW

File: plugin/src/buildWorkletString.ts [hunk 1]
```diff
@@ -84,7 +86,7 @@ export function buildWorkletString(
 
   const transformed = transformSync(code, {
     plugins: [prependClosureVariablesIfNecessary(closureVariables)],
-    compact: !includeSourceMap,
+    compact: true,
```
Comment (tomekzaw) [inline plugin/src/buildWorkletString.ts:89]: It's too late to think, please somebody confirm if we can leave `true` here.

Comment (piaskowyk) [reply plugin/src/buildWorkletString.ts:89]: If I remember correctly, this change was required for proper source mapping for Hermes debugger. When we use `compact: true` then some debugger breakpoints will be shifted by some random number of lines.

File: __tests__/plugin.test.ts [hunk 2]
```diff
@@ -72,7 +72,21 @@ describe('babel plugin', () => {
...
+
+      const input = html`<script>
+        function foo() {
+          'worklet';
+          var foo = 'bar';
+        }
+      </script>`;
+
+      const { code } = runPlugin(input, { sourceMaps: true });
+      expect(code).toContain('{\\"version\\":3');
```
Comment (piaskowyk) [inline __tests__/plugin.test.ts:89]: Why do we check here if the `version` property exists instead of checking that the `sourceMap` property exists?

File: __tests__/plugin.test.ts
Comment (piaskowyk) [inline __tests__/plugin.test.ts:1]: Now we are testing debug version of the plugin, maybe we should also add tests for the release version of the plugin. What do you think?

Comment (tjzel) [reply __tests__/plugin.test.ts:1]: There are tests that compare selected behaviours for both environments.

File: plugin/src/buildWorkletString.ts [hunk 3]
```diff
@@ -31,6 +31,8 @@ import * as fs from 'fs';
 import { isRelease } from './utils';
 import { WorkletizableFunction } from './types';
 
+const MOCK_SOURCE_MAP = 'mock source map';
```
Comment (tjzel) [inline plugin/src/buildWorkletString.ts:34]:
```suggestion
const MOCKED_SOURCE_MAP = 'mocked source map';
```

**Review by tjzel — CHANGES_REQUESTED** [review]

File: __tests__/plugin.test.ts (hunk 2 above)
Comment (tomekzaw) [reply __tests__/plugin.test.ts]: Changed in 47e279f8d13fb32ac4c17cc99d999c19ab4b4b7f

File: __tests__/plugin.test.ts
Comment (piaskowyk) [reply __tests__/plugin.test.ts:1]: I can't find them. Could you send me a link to this test?

Comment (tjzel) [reply __tests__/plugin.test.ts:1]: https://github.com/software-mansion/react-native-reanimated/blob/main/__tests__/plugin.test.ts#L1386

not many of them so I guess this is a valid concern. 

We also have things like this
https://github.com/software-mansion/react-native-reanimated/blob/main/__tests__/plugin.test.ts#L75

which just need their respective counterpart.

Comment (piaskowyk) [reply __tests__/plugin.test.ts:1]: Thanks! Glad to hear that we have test to check it but I think we could add a couple next of test cases. But probably this is out of this PR so we can do it on another.

File: __tests__/plugin.test.ts [hunk 4]
```diff
@@ -62,17 +62,32 @@ describe('babel plugin', () => {
...
           var foo = 'bar';
         }
       </script>`;
 
-      const { code } = runPlugin(input, {});
-      expect(code).toContain(`f.__version = "${packageVersion}";`);
+      const { code } = runPlugin(input);
+      expect(code).toContain('sourceMap: "{');
+      expect(code).toContain(
+        '\\"mappings\\":\\"AACQ,SAAAA,GAASA,CAAA,CAAG,CAEV,GAAI,CAAAA,GAAG,CAAG,KAAK,CACjB\\"'
```
Comment (tjzel) [inline __tests__/plugin.test.ts]: A comment on this line would be nice.

File: plugin/src/buildWorkletString.ts (hunk 3 above)
Comment (tjzel) [reply plugin/src/buildWorkletString.ts:34]: Actually nevermind, everywhere else we just have MOCK

File: __tests__/plugin.test.ts [hunk 5]
```diff
@@ -821,7 +836,7 @@ describe('babel plugin', () => {
       </script>`;
 
       const { code } = runPlugin(input);
-      expect(code).toContain('...{b:2,c:3}');
+      expect(code).toContain('...{\\n      b: 2,\\n      c: 3\\n    }');
```
Comment (tjzel) [inline __tests__/plugin.test.ts:839]: why is this necessary? it was working previously

File: __tests__/plugin.test.ts [hunk 6]
```diff
@@ -62,17 +62,32 @@ describe('babel plugin', () => {
     });
 
     it('injects its version', () => {
```
Comment (tjzel) [inline __tests__/plugin.test.ts:64]: Some of those tests could be moved to 'for debugging' section. Also we could add a test that checks for version not being present in release builds.

**Review by tjzel — CHANGES_REQUESTED** [review]

File: plugin/src/buildWorkletString.ts (hunk 1 above)
Comment (tomekzaw) [reply plugin/src/buildWorkletString.ts:89]: Turns out we already effectively use `compact: true` when running an app. I've just checked it and breakpoint seem to work fine after this change.

**Review by tjzel — APPROVED** [review]

**Review by piaskowyk — APPROVED** [review]
